src/Plugin.php

```diff
@@ -98,4 +118,68 @@ public function patchPackage(PackageEvent $event): void {
     parent::patchPackage($event);
   }
 
+  /**
+   * Check if package versions have changed since the dependent patches lock was created.
+   * Only returns true once per process to avoid multiple regenerations.
+   */
+  protected function hasPackageVersionsChanged(): bool {
+    // If we've already regenerated patches in this process, don't do it again
+    if ($this->patchesRegenerated) {
+      return false;
+    }
+
+    try {
+      $lockData = $this->locker->getLockData();
+    } catch (\Exception $e) {
+      return true;
+    }
+
+    if (!isset($lockData['_composer_lock_hash'])) {
+      return true;
+    }
+
+    // Use a simple heuristic: check if the patches lock file is newer than composer.lock
+    // If composer.lock is newer, then versions have likely changed
+    $composerLockPath = $this->composer->getConfig()->get('vendor-dir') . '/../composer.lock';
+    $patchesLockPath = static::getPatchesLockFilePath();
+    
+    if (!file_exists($composerLockPath) || !file_exists($patchesLockPath)) {
+      return true;
+    }
+    
+    $composerLockTime = filemtime($composerLockPath);
+    $patchesLockTime = filemtime($patchesLockPath);
+    
+    // If composer.lock was modified after patches lock, versions likely changed
+    return $composerLockTime > $patchesLockTime;
```

Comment:
I think this part can be simplified. Also, not sure what's the point of storing the composer.lock content-hash in the plugin's lock, if we then don't use it here and check file modification times instead.

When I tested, this also worked, and the dependent-patches.lock was correctly updated when changing the version constraints of the patched package in composer.json and running composer update.

```suggestion
    return $lockData['_composer_lock_hash'] !== $this->getComposerLockHash();
```